crates/orchestrator/src/jobs/mod.rs

```diff
@@ -333,8 +333,8 @@ pub async fn verify_job(id: Uuid, config: Arc<Config>) -> Result<(), JobError> {
     tracing::Span::current().record("internal_id", job.internal_id.clone());
 
     match job.status {
-        JobStatus::PendingVerification => {
-            tracing::debug!(job_id = ?id, "Job status is PendingVerification, proceeding with verification");
+        JobStatus::PendingVerification | JobStatus::VerificationTimeout => {
```

Comment:
As discussed, let's add an `IMP!` comment here explaining why doing this is fine, 
referring that we don't add the jobs that timeout to the queue back again